drop-storage/src/types.rs

```diff
@@ -0,0 +1,192 @@
+use serde::Serialize;
+
+type TransferId = String;
+type FilePath = String;
+
+#[derive(Debug, Copy, Clone)]
+pub enum TransferType {
+    Incoming = 0,
+    Outgoing = 1,
+}
+
+impl From<TransferType> for i32 {
+    fn from(transfer_type: TransferType) -> Self {
+        match transfer_type {
+            TransferType::Incoming => 0,
+            TransferType::Outgoing => 1,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct TransferPath {
+    pub id: String,
+    pub path: String,
+    pub size: i64,
+}
+
+#[derive(Debug)]
+pub struct TransferInfo {
+    pub id: String,
+    pub peer: String,
+    pub files: Vec<TransferPath>,
+}
+
+#[derive(Debug)]
+pub enum Event {
+    Pending(TransferType, TransferInfo),
+    Started(TransferType, TransferId, FilePath),
+
+    FileCanceled(TransferType, TransferId, FilePath, bool),
+    TransferCanceled(TransferType, TransferInfo, bool),
+
+    FileFailed(TransferType, TransferId, FilePath, u32),
+    TransferFailed(TransferType, TransferInfo, u32),
+
+    FileUploadComplete(TransferId, FilePath),
+    FileDownloadComplete(TransferId, FilePath, String),
+
+    Progress(TransferId, FilePath, i64),
```

Comment:
Since we're writing the events from scratch, I would suggest not using things like `bool`, `u32`. It's way too confusing. We should either make named fields inside the enum variants or use a newtypes

Reply:
fixed